pkg/util/util.go

```diff
@@ -217,13 +218,36 @@ func GetBrokerConfig(broker v1beta1.Broker, clusterSpec v1beta1.KafkaClusterSpec
 		bConfig = broker.BrokerConfig.DeepCopy()
 	}
 
-	err := mergo.Merge(bConfig, clusterSpec.BrokerConfigGroups[broker.BrokerConfigGroup], mergo.WithAppendSlice)
+	groupConfig, exists := clusterSpec.BrokerConfigGroups[broker.BrokerConfigGroup]
+	if !exists {
+		return nil, errors.NewWithDetails("missing brokerConfigGroup", "key", broker.BrokerConfigGroup)
+	}
+
+	dstAffinity := &corev1.Affinity{}
+	srcAffinity := &corev1.Affinity{}
+
+	if groupConfig.Affinity != nil {
+		dstAffinity = groupConfig.Affinity.DeepCopy()
+	}
+	if bConfig.Affinity != nil {
+		srcAffinity = bConfig.Affinity
+	}
+
+	if err := mergo.Merge(dstAffinity, srcAffinity, mergo.WithOverride); err != nil {
+		return nil, errors.WrapIf(err, "could not merge brokerConfig.Affinity with ConfigGroup.Affinity")
+	}
+
+	err := mergo.Merge(bConfig, groupConfig, mergo.WithAppendSlice)
 	if err != nil {
 		return nil, errors.WrapIf(err, "could not merge brokerConfig with ConfigGroup")
 	}
 
 	bConfig.StorageConfigs = dedupStorageConfigs(bConfig.StorageConfigs)
 
+	if !(groupConfig.Affinity == nil && bConfig.Affinity == nil) {
```

Comment:
This doesn't need parenthesis, but it's up to you, which one looks better :) 
```suggestion
	if groupConfig.Affinity != nil || bConfig.Affinity != nil {
```